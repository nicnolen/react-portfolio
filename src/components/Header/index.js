import React, { useState } from 'react';
import { Routes, Route, HashRouter } from 'react-router-dom';
import Project from '../Portfolio';
import About from '../About';
import Navigation from '../Navigation';
import Resume from '../Resume';
import Contact from '../Contact';

function Header() {
  // conditionally render the Contact component
  const [aboutSelected, setAboutSelected] = useState(true);
  // set categories
  const [categories] = useState([
    {
      name: 'project',
      description:
        'Listed out my top six projects with links to the deployed link and the GitHub link',
    },
    {
      name: 'contact',
      description: 'Form to contact me',
    },
    {
      name: 'resume',
      description:
        'Link to download my resume as well as a list of my skills with links to the documentation for the front end and back end skills',
    },
  ]);

  const [currentCategory, setCurrentCategory] = useState(categories[0]);
  return (
    <header>
      <HashRouter>
        <nav>
          <Navigation
            categories={categories}
            setCurrentCategory={setCurrentCategory}
            currentCategory={currentCategory}
            aboutSelected={aboutSelected}
            setAboutSelected={setAboutSelected}
          />
        </nav>

        <div>
          <Routes>
            <Route exact path='/' element={<About />} />
            <Route path='/about' element={<About />}></Route>
            <Route path='/project' element={<Project />}></Route>
            <Route path='/contact' element={<Contact />}></Route>
            <Route path='/resume' element={<Resume />}></Route>
          </Routes>
        </div>
      </HashRouter>
      <div>
        <About />
        <Project />
        <Resume />
        <Contact />
      </div>
    </header>
  );
}

export default Header;
