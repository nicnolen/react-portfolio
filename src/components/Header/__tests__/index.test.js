// Import dependencies
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '..';

// After each test, make sure there isnt any leftover memory that could give you false results
afterEach(cleanup);

// Declare the components you are testing
describe('Header component', () => {
  // verify the component is rendering properly
  it('Header component renders', () => {
    render(<Header />);
  });

  // compare snapshot versions of the DOM node structure
  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});