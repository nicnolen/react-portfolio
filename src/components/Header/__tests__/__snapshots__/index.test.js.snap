// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="navbar navbar-expand-lg navbar-light bg-secondary sticky-top"
    >
      <div
        class="justify-content-start container"
      >
        <a
          class="navbar-brand"
          href="/react-portfolio/"
        >
          <h1
            class="text-light nav-title-font mx-4"
          >
            Nicholas Nolen
          </h1>
        </a>
      </div>
      <div
        class="justify-content-end container"
      >
        <ul
          class="navbar-nav mx-4 navitem-indent"
        >
          <li
            class="mx-2 nav-item"
          >
            <a
              class="navLink mx-2 navActive"
              href="/about"
            >
              <span>
                About Me
              </span>
            </a>
          </li>
          <li
            class="mx-1 nav-item"
          >
            <a
              class="navLink mx-2 false"
              href="/project"
            >
              <span>
                Project
              </span>
            </a>
          </li>
          <li
            class="mx-1 nav-item"
          >
            <a
              class="navLink mx-2 false"
              href="/contact"
            >
              <span>
                Contact
              </span>
            </a>
          </li>
          <li
            class="mx-1 nav-item"
          >
            <a
              class="navLink mx-2 false"
              href="/resume"
            >
              <span>
                Resume
              </span>
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div />
  </header>
</DocumentFragment>
`;
